src/server.cpp

```diff
@@ -2497,11 +2497,21 @@ bool Server::addMediaFile(const std::string &filename,
 	}
 	// Ok, attempt to load the file and add to cache
 
-	// Read data
-	std::string filedata;
-	if (!fs::ReadFile(filepath, filedata)) {
+	// Read data if not given
+	bool ok = true;
+	std::string filedata_file;
+	std::string &filedata = [&]() -> std::string & {
+		if (filedata_to && !filedata_to->empty()) {
```

Comment:
What if someone uses `data = ""`? Won't that cause `addMediaFile` to try to read from `filepath`? IMO you should add a separate parameter to signal that `filedata_to` is an input, rather than checking whether it is empty. `data = ""` should be supported out of principle.